Making grammar unambiguous (#99)
Materials-Consortia/OPTiMaDe: optimade.md

>  OPTiMaDe API specification v0.9.8-develop

Reviewer: Is this intentional?

Author: No... Probably an accidental change.

Author: Reinserted the '#' character back.

The following tokens are used in the filter query component:

* **Property names** (see section [6. Entry List](#h.6)): Are to follow the identifier

>   syntax of programming languages -- the first character MUST be a lowercase letter, the subsequent symbols

Reviewer 2: Isn't it a bit weird now to keep the "follow the identifier syntax of programming languages", when most programming language allows mixed-case identifiers?

Author: I agree, it may be more confusing than clarifying. Probably we should remove reference to "most programming languages" at all.

Author: Re-wrote the "most programming languages" paragraph.

> IdentifierZipAddon = ':', Identifier, {':', Identifier} ;

Reviewer 2: Why are these ':' not Colon?

Digit = '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9' ;

(* White-space: *)

(* Special character tokens: *)

tab = ? \t ?;
nl  = ? \n ?;
cr  = ? \r ?;
vt  = ? \v ?;
ff  = ? \f ?;

Space = ' ' | tab | nl | cr | vt | ff ;

Spaces = Space, { Space } ;

(* The 'UnicodeHighChar' specifies all Unicode characters above 0x7F;
   the syntax used is the onw compatible with Grammatica: *)

UnicodeHighChar = ? [^\x00-\x7F] ? ;
 
(* END EBNF GRAMMAR Number *)
(* END EBNF GRAMMAR Filter *)
```

> Note: whe implementing a parser according this grammar, the

Reviewer 2: ```suggestion
Note: when implementing a parser according this grammar, the
```

implementers MAY choose to construct a lexer that ignores all

> whitespace (space, tabs, newlines, vertical tabulation an format feed

Reviewer 2: ```suggestion
whitespace (space, tabs, newlines, vertical tabulation and form feed
```

characters, as described in the grammar 'Space' definition), and use

> such lexer to recognise language elements that are described in the

Reviewer 2: ```suggestion
such a lexer to recognize language elements that are described in the
```
